src/lxc/log.c

```diff
@@ -170,10 +170,13 @@ static int log_append_logfile(const struct lxc_log_appender *appender,
 		     event->locinfo->file, event->locinfo->func,
 		     event->locinfo->line);
 
-	n += vsnprintf(buffer + n, sizeof(buffer) - n, event->fmt,
-		       *event->vap);
+	if (n < 0)
+		return n;
 
-	if (n >= sizeof(buffer) - 1) {
+	if (n < sizeof(buffer) - 1)
```

Comment:
It maybe smarter to only insert the check for `n < 0` but then still call `vsnprintf()` unconditionally as it is done now and let it truncate the string. This way we provide a little more information but also warn the user that happened.